### ISSUE TITLE
Albumentation no effect in ultralytics

### ISSUE THREAD

**Issue by lukebor** [ultralytics/ultralytics#26130]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I was testing some augmentations in my pipeline so i changed augment.py set p=1 and run training. Watche train_batches.jpg and no effect. Then i made a test and run training with checking 1st epoch losses and metrics. Whatever i change in probabilities and augmentations i got always same effect. It was working perfectly in previous versions, i dont know when it has changed. only thing i changed from my pipeline was adding 
channels: 1 to my data.yaml 
yolo log correctly shows my augmentations in line albumentations: ....

tested on 1 GPU and 4 GPU, same effect

### Environment

albumentations: OneOf([
  ColorJitter(p=0.3, brightness=(0.7, 1.4), contrast=(0.7, 1.4), hue=(-0.5, 0.5), saturation=(0.8, 1.2)),
  RandomToneCurve(p=0.3, per_channel=False, scale=0.3),
  RandomGamma(p=0.3, gamma_limit=(80.0, 120.0)),
], p=0.5)

changing p and augmentations has no effect on loss and metrics

### Minimal Reproducible Example

yolo train cfg=my.yaml data=data.yaml model=yolo26n.pt epochs=50 imgsz=640 device=0 batch=64 workers=10

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lukebor, thank you for reporting that changing Albumentations probabilities produces identical `train_batch` visualizations, losses, and metrics when using `channels: 1`. This is an automated response, and an Ultralytics engineer will assist soon.

Please provide the installed `ultralytics`, `albumentations`, Python, and PyTorch versions, along with the exact `augment.py` changes and one generated `train_batch*.jpg` comparison with augmentations disabled and enabled. This will help us investigate whether the behavior is specific to the grayscale channel configuration or the augmentation pipeline.

Please also upgrade to the latest `ultralytics` package and verify whether the issue persists:

```bash
pip install -U ultralytics
```

If it does, a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) will help us debug it.

**Comment by onuralpszr** [comment]
Hello @lukebor I made PR and waiting for approval so you can check and try as well.

**Comment by lukebor** [comment]
Hello @onuralpszr  yes now it works, thank you . Didn't see this 3-channel allowance

**Comment by onuralpszr** [comment]
> Hello @onuralpszr  yes now it works, thank you . Didn't see this 3-channel allowance

Glad to hear it. :)

**Comment by UltralyticsAssistant** [comment]
A potential fix is now available in the [merged pull request](https://github.com/ultralytics/ultralytics/pull/26133) by @onuralpszr with contributions from @glenn-jocher.

Updated Albumentations preprocessing to run on both grayscale and three-channel images instead of only color images.

Please test the merged change using this repository's documented workflow. If the issue persists, share the updated behavior and any new diagnostic details. Thank you for reporting it! 🙏
